dspace-api/src/main/java/org/dspace/administer/StructBuilder.java

```diff
@@ -171,35 +250,154 @@ public static void main(String[] argv)
         Element[] elements = handleCommunities(context, first, null);
 
         // generate the output
-        Element root = xmlOutput.getRootElement();
-        for (int i = 0; i < elements.length; i++) {
-            root.addContent(elements[i]);
+        final Element root = new Element(RESULT_ROOT);
+
+        for (Element element : elements) {
+            root.addContent(element);
         }
 
-        // finally write the string into the output file
+        // finally write the string into the output file.
+        final org.jdom.Document xmlOutput = new org.jdom.Document(root);
         try {
-            BufferedWriter out = new BufferedWriter(new FileWriter(output));
-            out.write(new XMLOutputter().outputString(xmlOutput));
-            out.close();
+            new XMLOutputter().output(xmlOutput, output);
```

Comment:
In the export method we use `Format.getPrettyFormat()` to indent the XML. Why don't we use this for the xml output of the import method?

Reply:
That sounds like a good idea, but it seems out-of-scope for the present patch and issue which are focused on the new export feature.  I think that a new issue and a small patch would be welcome.